Update linters and python version for linting in CI (#15200)
python/mypy: CONTRIBUTING.md

hash -r  # This resets shell PATH cache, not necessary on Windows
```

> **Note**

> > Some code quality tools are not compatible with old python versions, you'll
> > need python version 3.8 or higher to install and run them.

Reviewer: Slightly shorter:

```suggestion
> You'll need python version 3.8 or higher to install all mypy's test requirements.
```

Author: Sure it is shorter but it is a little bit ambiguous. You can run all tests perfectly on python3.7 but you cannot run the linters. I don't have a strong opinion though, this only meant to clarify potential confusion. Feel free to commit the suggestion if you like.

Reviewer: Well, I think keeping `CONTRIBUTING.md` concise is important; an overly long `CONTRIBUTING` can be intimidating to new contributors :)

I also think the important information is "you need Python 3.8+", so we should definitely say that first and explain afterwards. At the moment, you have the explanation ("some code quality tools are not compatible with old Python versions") before the important information.

Maybe this is less ambiguous?

```suggestion
> You'll need Python 3.8 or higher to install all requirements listed in
> test-requirements.txt
```

> Feel free to commit the suggestion if you like.

It's not in my power to do so; I'm a maintainer at typeshed, but over here I'm only a triager :)
